### PR TITLE
[Config] change some static config to dynamic config and delete some unused config

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

1. Change some static config of BE to dynamic config so that we can reset some parameters without restart BE.
2. Delete some unused BE config.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

**Review by caiconghui — DISMISSED** [review]
+1

File: be/src/common/config.h [hunk 1]
```diff
@@ -256,21 +222,18 @@ CONF_Int32(file_descriptor_cache_capacity, "32768");
 // modify them upon necessity
 CONF_Int32(min_file_descriptor_number, "60000");
 CONF_Int64(index_stream_cache_capacity, "10737418240");
-// CONF_Int64(max_packed_row_block_size, "20971520");
 
 // Cache for storage page size
 CONF_String(storage_page_cache_limit, "20%");
 // whether to disable page cache feature in storage
-CONF_Bool(disable_storage_page_cache, "false");
+CONF_mBool(disable_storage_page_cache, "false");
```
Comment (acelyc111) [inline be/src/common/config.h:229]: There is a case that `disable_storage_page_cache ` is true at first, then changed to false, will the memory used by cache has no chance to free?

File: be/src/common/config.h [hunk 2]
```diff
@@ -424,8 +378,7 @@ CONF_Bool(enable_quadratic_probing, "false");
 CONF_String(pprof_profile_dir, "${DORIS_HOME}/log");
 
 // for partition
-// CONF_Bool(enable_partitioned_hash_join, "false")
-CONF_Bool(enable_partitioned_aggregation, "true");
+CONF_mBool(enable_partitioned_aggregation, "true");
```
Comment (acelyc111) [inline be/src/common/config.h:381]: There is a memory leak risk, `enable_partitioned_aggregation` is true at first and then changed to false, memory allocated has no chance to free.

File: be/src/common/config.h [hunk 3]
```diff
@@ -472,7 +412,7 @@ CONF_String(buffer_pool_clean_pages_limit, "20G");
 CONF_mInt64(memory_maintenance_sleep_time_s, "10");
 
 // Alignment
-CONF_Int32(memory_max_alignment, "16");
+CONF_mInt32(memory_max_alignment, "16");
```
Comment (acelyc111) [inline be/src/common/config.h:415]: I think you can remove this config, and define it as a static const value, there is no reason to update it.

File: be/src/common/config.h [hunk 4]
```diff
@@ -482,8 +422,8 @@ CONF_mInt64(write_buffer_size, "104857600");
 // NOTICE(cmy): set these default values very large because we don't want to
 // impact the load performance when user upgrading Doris.
 // user should set these configs properly if necessary.
-CONF_Int64(load_process_max_memory_limit_bytes, "107374182400"); // 100GB
-CONF_Int32(load_process_max_memory_limit_percent, "80");         // 80%
+CONF_mInt64(load_process_max_memory_limit_bytes, "107374182400"); // 100GB
+CONF_mInt32(load_process_max_memory_limit_percent, "80");         // 80%
```
Comment (acelyc111) [inline be/src/common/config.h:426]: These 2 configs are only changed in LoadChannelMgr's init function, which will only be called only once when server start, they are immutable.

File: be/src/common/config.h (hunk 1 above)
Comment (weizuo93) [reply be/src/common/config.h:229]: The config `disable_storage_page_cache`is not used to close cache but to choose whether to use the cache during read.  So the memory used by cache will not be freed.

File: be/src/common/config.h (hunk 3 above)
Comment (weizuo93) [reply be/src/common/config.h]: OK.

File: be/src/common/config.h (hunk 4 above)
Comment (weizuo93) [reply be/src/common/config.h]: OK.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**